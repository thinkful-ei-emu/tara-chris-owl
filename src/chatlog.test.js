import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './chatlog.js';
import renderer from 'react-test-renderer';

describe('Participant Component', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<ChatLog participants={[{
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    }]} chatEvents={[  {
      participantId: 3,
      type: 'join-stage',
      timestamp: 1548852664247
  }]}/>, div);

        ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
      const tree = renderer
        .create(<ChatLog participants={[{
          id: 1,
          name: 'Koren Templeton',
          avatar:
              'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
      }]} chatEvents={[  {
        participantId: 3,
        type: 'join-stage',
        timestamp: 1548852664247
    }]}/>)
        .toJSON();
      expect(tree).toMatchSnapshot();  
      });
})