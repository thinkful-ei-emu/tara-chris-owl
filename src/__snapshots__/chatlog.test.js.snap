// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component renders the UI as expected 1`] = `
<div
  className="chatLog"
>
  <div
    className="chatBox"
  >
    <p>
      <em />
       joined the stage
    </p>
  </div>
</div>
`;
